### ISSUE TITLE
decoder: Segmentation fault

### ISSUE THREAD

**Issue by jindyliu** [facebookresearch/wav2letter#296]
I was able to run Test binary and create Emission set "#root#volume#e2e_data_speech#test-other.lst.bin" .

I am trying to run Decoder now by using Emission set. Here is my command and the output I am getting.

> [root@Tencent-SNG decode_w2l_e2e]# /root/wav2letter/build/Decoder --flagsfile=/root/volume/decode_w2l_e2e/decode_clean_emission.cfg
I0515 09:32:53.703246 30919 Decode.cpp:52] Parsing command line flags
I0515 09:32:53.703351 30919 Decode.cpp:56] Reading flags from file /root/volume/decode_w2l_e2e/decode_clean_emission.cfg
I0515 09:32:53.703531 30919 Decode.cpp:95] [Serialization] Loading file: /root/volume/decode_w2l_e2e/#root#volume#e2e_data_speech#test-other.lst.bin
I0515 09:32:53.853431 30919 Decode.cpp:106] Gflags after parsing
--adambeta1=0.90000000000000002; --adambeta2=0.999; --am=/root/volume/decode_w2l_e2e/epoch45-ls4_50/model.bin; --arch=tiny.arch; --archdir=; --attention=content; --attnWindow=no; --attnconvchannel=0; --attnconvkernel=0; --attndim=0; --batchsize=16; --beamscore=25; --beamsize=1000; --channels=1; --criterion=asg; --critoptim=sgd; --datadir=; --dataorder=input; --devwin=0; --emission_dir=/root/volume/decode_w2l_e2e/; --enable_distributed=true; --encoderdim=0; --eostoken=false; --everstoredb=false; --fftcachesize=1; --filterbanks=40; --flagsfile=/root/volume/decode_w2l_e2e/decode_clean_emission.cfg; --forceendsil=false; --gamma=1; --garbage=false; --input=flac; --inputbinsize=100; --inputfeeding=false; --iter=1000000; --itersave=false; --labelsmooth=0; --leftWindowSize=50; --lexicon=/root/volume/decode_w2l_e2e/enwiki-2018-04-4gram/enwiki-lexicon; --linlr=-1; --linlrcrit=-1; --linseg=1; --listdata=true; --lm=/root/volume/decode_w2l_e2e/enwiki-2018-04-4gram/enwiki-4gram.bin; --lmtype=kenlm; --lmweight=2.5; --localnrmlleftctx=0; --localnrmlrightctx=0; --logadd=false; --lr=0.59999999999999998; --lrcrit=0.059999999999999998; --maxdecoderoutputlen=200; --maxgradnorm=0.20000000000000001; --maxisz=30000; --maxload=-1; --maxrate=10; --maxsil=50; --maxtsz=9223372036854775807; --maxword=-1; --melfloor=1; --memstepsize=10485760; --mfcc=false; --mfcccoeffs=13; --mfsc=true; --minisz=0; --minrate=3; --minsil=0; --mintsz=0; --momentum=0.5; --netoptim=sgd; --noresample=false; --nthread=1; --nthread_decoder=1; --onorm=target; --optimepsilon=1e-08; --optimrho=0.90000000000000002; --outputbinsize=5; --pctteacherforcing=100; --pcttraineval=100; --pow=false; --pretrainWindow=0; --replabel=0; --reportiters=1000; --rightWindowSize=50; --rndv_filepath=; --rundir=; --runname=model_librispeech_tiny_sa; --samplerate=16000; --sampletarget=0; --samplingstrategy=rand; --sclite=/root/volume/e2e_data_speech/decode_e2e_log/; --seed=0; --show=true; --showletters=true; --silweight=-0.40000000000000002; --smearing=max; --softwoffset=10; --softwrate=5; --softwstd=5; --sqnorm=true; --stepsize=1000000; --surround=|; --tag=; --target=tkn; --test=/root/volume/e2e_data_speech/test-other.lst; --tokens=tokens.txt; --tokensdir=/root/volume/decode_w2l_e2e/epoch45-ls4_50; --train=librispeech/train-all.lst; --trainWithWindow=false; --transdiag=4; --unkweight=-inf; --usewordpiece=false; --valid=common_voice/clips_5p_test.lst,tedlium/test.lst,tatoeba/test.lst,librispeech/test-clean.lst,librispeech/test-other.lst,speech_commands/test.lst; --weightdecay=0; --wordscore=1; --wordseparator=|; --world_rank=0; --world_size=1; --flagfile=; --fromenv=; --tryfromenv=; --undefok=; --tab_completion_columns=80; --tab_completion_word=; --help=false; --helpfull=false; --helpmatch=; --helpon=; --helppackage=false; --helpshort=false; --helpxml=false; --version=false;
I0515 09:32:53.853808 30919 Decode.cpp:112] Number of classes (network): 28
I0515 09:32:55.617547 30919 Decode.cpp:116] Number of words: 677264
I0515 09:32:56.242547 30919 Decode.cpp:165] [Dataset] Number of samples per thread: 2932
I0515 09:32:56.394556 30919 Decode.cpp:243] [Decoder] LM constructed.
I0515 09:32:57.465101 30919 Trie.cpp:48] [Trie] Trie label number reached limit: 6
I0515 09:32:57.734711 30919 Trie.cpp:48] [Trie] Trie label number reached limit: 6
I0515 09:32:57.868026 30919 Trie.cpp:48] [Trie] Trie label number reached limit: 6
I0515 09:32:57.897307 30919 Trie.cpp:48] [Trie] Trie label number reached limit: 6
I0515 09:32:58.144539 30919 Trie.cpp:48] [Trie] Trie label number reached limit: 6
I0515 09:32:58.150918 30919 Trie.cpp:48] [Trie] Trie label number reached limit: 6
I0515 09:32:58.629441 30919 Trie.cpp:48] [Trie] Trie label number reached limit: 6
I0515 09:32:58.678337 30919 Trie.cpp:48] [Trie] Trie label number reached limit: 6
I0515 09:32:58.738597 30919 Trie.cpp:48] [Trie] Trie label number reached limit: 6
I0515 09:32:58.976301 30919 Trie.cpp:48] [Trie] Trie label number reached limit: 6
I0515 09:32:58.978530 30919 Decode.cpp:276] [Decoder] Trie planted.
I0515 09:32:59.417084 30919 Decode.cpp:288] [Decoder] Trie smeared.
I0515 09:32:59.417382 30920 Decode.cpp:297] [Decoder] Decoder loaded in thread: 0
*** Aborted at 1557912779 (unix time) try "date -d @1557912779" if you are using GNU date ***
PC: @           0x536c55 w2l::Decoder::candidatesAdd()
*** SIGSEGV (@0x79) received by PID 30919 (TID 0x7f8555994700) from PID 121; stack trace: ***
    @     0x7f85ae1a95d0 (unknown)
    @           0x536c55 w2l::Decoder::candidatesAdd()
    @           0x53a00d w2l::Decoder::decodeContinue()
    @           0x53bb51 w2l::Decoder::decode()
    @           0x471c9c _ZZ4mainENKUliiiE2_clEiii
    @           0x47afea std::_Function_handler<>::_M_invoke()
    @           0x47b07e std::__future_base::_State_base::_M_do_set()
    @     0x7f85ae1a6e40 __GI___pthread_once
    @           0x4707ba _ZNSt13__future_base11_Task_stateISt5_BindIFZ4mainEUliiiE2_iiiEESaIiEFvvEE6_M_runEv
    @           0x4716bf _ZNSt17_Function_handlerIFvvEZN2fl10ThreadPool7enqueueIRZ4mainEUliiiE2_IRiS6_S6_EEESt6futureINSt9result_ofIFT_DpT0_EE4typeEEOS9_DpOSA_EUlvE_E9_M_invokeERKSt9_Any_data
    @           0x47da76 _ZZN2fl10ThreadPoolC1EmRKSt8functionIFvmEEENKUlvE_clEv
    @     0x7f8580e10060 execute_native_thread_routine_compat
    @     0x7f85ae1a1dd5 start_thread
    @     0x7f85769d0ead __clone
Segmentation fault

How to slove the Segmentation fault error ?

**Comment by xuqiantong** [comment]
Hi, it's hard to reproduce it and see where the segfault happens. Can you share some of your local changes or the core dump stack trace?

**Comment by xuqiantong** [comment]
No following questions. closing.

**Comment by 00001101-xt** [comment]
Hi, @xuqiantong I encountered a similar problem and here is my log:

```
I1108 09:58:34.905297 36555 Decode.cpp:133] Number of classes (network): 114
I1108 09:58:35.307626 36555 Decode.cpp:140] Number of words: 56597
I1108 09:58:35.550278 36555 W2lListFilesDataset.cpp:137] 20 files found. 
I1108 09:58:35.550333 36555 Utils.cpp:102] Filtered 0/20 samples
I1108 09:58:35.550487 36555 W2lListFilesDataset.cpp:62] Total batches (i.e. iters): 20
I1108 09:58:35.550523 36555 Decode.cpp:154] [Serialization] Running forward pass ...
I1108 09:58:40.037379 36555 Decode.cpp:201] [Dataset] Number of samples per thread: 20
I1108 09:58:41.480674 36555 Decode.cpp:308] [Decoder] LM constructed.
[Trie] Trie label number reached limit: 6
...
[Trie] Trie label number reached limit: 6
I1108 09:58:42.134409 36555 Decode.cpp:332] [Decoder] Trie planted.
I1108 09:58:42.229481 36555 Decode.cpp:344] [Decoder] Trie smeared.
I1108 09:58:42.239115 36555 Decode.cpp:413] [Decoder] Seq2Seq decoder with token-LM loaded in thread: 0
F1108 09:58:42.344645 36555 Decode.cpp:519] Exception in thread 0
Batched decoding does not support models with window
*** Check failure stack trace: ***
    @     0x7f02cee6ce6d  (unknown)
    @     0x7f02cee6eced  (unknown)
    @     0x7f02cee6ca5c  (unknown)
    @     0x7f02cee6f63e  (unknown)
    @           0x41c5fc  _ZZ4mainENKUliiiE2_clEiii
    @           0x41c66f  _ZZ4mainENKUlvE3_clEv
    @           0x41f091  main
    @     0x7f02ce089b15  __libc_start_main
    @           0x41a565  (unknown)
Aborted (core dumped)
```
